core/src/main/java/org/elasticsearch/index/mapper/Mapping.java

```diff
@@ -98,7 +98,8 @@ public Mapping merge(Mapping mergeWith, boolean updateAllTypes) {
             }
             mergedMetaDataMappers.put(merged.getClass(), merged);
         }
-        return new Mapping(indexCreated, mergedRoot, mergedMetaDataMappers.values().toArray(new MetadataFieldMapper[0]), mergeWith.meta);
+        return new Mapping(indexCreated, mergedRoot, mergedMetaDataMappers.values().toArray(new MetadataFieldMapper[0]),
+            mergeWith.meta != null ? mergeWith.meta : meta);
```

Comment:
I would flip this logic to remove negations. ie `mergeWith.meta == null ? meta : mergeWith.meta`. Positive logic is easier to reason about. It might also be better to just move it out to a local instead of trying to inline it.

Reply:
Done